src/components/Redirect.jsx

```diff
@@ -43,30 +50,49 @@ type RedirectProps = {
  * <Router to='http://example.com' />
  * // external URL, URL object
  * <Router to={new URL('http://example.com')} />
+
+ * // internal route, permanent redirect
+ * <Router to='http://example.com' permanent />
+ * // external URL, permanent redirect
+ * <Router to='http://example.com' permanent />
  * ```
  */
 class Redirect extends React.Component {
 	props: RedirectProps;
 	render() {
-		const { to, push } = this.props;
+		const { to, push, permanent } = this.props;
 		if (to instanceof URL || testForExternal(to)) {
+			// this is an external URL, so let `withSideEffect` handlers do redirect
 			return null;
 		}
-		return <RouterRedirect to={to} push={push} />;
+		// internal route URL
+		return (
```

Comment:
Could you explain this a bit? Sorry. Trying to understand shy you need to add a route to do the redirect inside of it.